### PR TITLE
Pulling code via clone & running CK on different releases using TAGS

### PR DESCRIPTION
**Pull request by alasifer**

### CODE REVIEW

File: src/matric/ck.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from src.utils import Subprocess
+from src.utils import copy_files
+from .index import MetricConfig
+
+
+class CK:
+    """
+    example:
+    java -jar ck-0.6.3-SNAPSHOT-jar-with-dependencies.jar /Users/ahmedalasifer/Desktop/FIT4003/scala
```
Comment (jeevanpillay0000) [inline src/matric/ck.py]: What does `/Users/ahmedalasifer/Desktop/FIT4003/scala` mean in this setting? The output dir?

File: src/__main__.py [hunk 2]
```diff
@@ -77,8 +78,8 @@ def main():
     os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "../.bigquery.gson"
 
     # init
-    app = AppBigQueryStorage()  # noqa: F841
-
+    # app = AppBigQueryStorage()  # noqa: F841
+    app = CloneRepo()
```
Comment (jeevanpillay0000) [inline src/__main__.py]: Great job adding a new app called CloneRepo!

File: src/matric/ck.py (hunk 1 above)
Comment (alasifer) [reply src/matric/ck.py]: The path in which the code to be analysed is located.

Comment (alasifer) [reply src/matric/ck.py]: the output in CK is stored under src, then i have to move it using mv

Comment (alasifer) [reply src/matric/ck.py]: but when we will be running SourceMeter, we can specify the output location

File: src/app/app.py [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
 
 
 class AppConfig(NamedTuple):
     name: str
     repository: Repository
     repository_config: RepositoryConfig
+    clone: Clone
+    clone_config: list
+    metric_config: MetricConfig
+    metric: CK
```
Comment (jeevanpillay0000) [inline src/app/app.py]: We need to clean this up.

Let's abstract this into these classes AppConfig, AppConfigDB, AppConfigGit with these properties
```
class AppConfig(NamedTuple):
     name: str
     metric_config: MetricConfig
     metric: CK

class AppConfigRepository(AppConfig):
     repository: Repository
     repository_config: RepositoryConfig

class AppConfigGit(AppConfig):
     clone: Clone:
     clone_config: list
```

File: src/app/app.py [hunk 4]
```diff
@@ -22,15 +32,22 @@ def __init__(self, appConfig: AppConfig):
...
+        self.logger.l.info("cloning to github using subprocess")
+        # repository = appConfig.repository
+        # repository_config = appConfig.repository_config
+        # self.logger.l.info(
+        #     "creating repository -> db_type: %s, api_type: %s",
+        #     RepositoryEnum.to_char(repository_config.dbType),
+        #     RepositoryBigQueryEnum.to_char(repository_config.apiType),
+        # )
+        # self.repository: Repository = repository(repository_config)
+        # self.logger.l.info("repository created!")
```
Comment (lpfloyd) [inline src/app/app.py:50]: We need to rework this too. I'll handle this.

File: src/matric/ck.py (hunk 1 above)
Comment (jeevanpillay0000) [reply src/matric/ck.py]: Btw, what do I have to install to get CK working? Could you add some info into README.md on setup guide.

File: src/app/app.py (hunk 4 above)
Comment (jeevanpillay0000) [inline src/app/app.py]: Rework needed.

Recommendations.
Create App class based on the AppConfig recommendations
e.g App, AppRepository, AppGit

File: src/app/app.py [hunk 5]
```diff
@@ -43,16 +60,33 @@ def Run(self):
...
-            if counter > 100000:
-                break
-            counter += 1
+        # print("The query data:")
+        # counter = 0
+        # for row in query_job:
+        #     print(row)
+        #     if counter > 100000:
+        #         break
+        #     counter += 1
```
Comment (jeevanpillay0000) [inline src/app/app.py]: Remove this.

File: src/app/app.py [hunk 6]
```diff
@@ -43,16 +60,33 @@ def Run(self):
...
+                    project_dir="{}/{}".format(
+                        new_repo.dir, new_repo.config.project_name
+                    ),
+                    output_dir="{}/{}".format(new_repo.dir, v),
+                )
+
+                m = self.metric(self.metric_config)
+                m.run_ck()
+                m.move_output(source=str(Path().resolve().parent) + "/src/")
+            remove_dir("{}/{}".format(new_repo.dir, new_repo.config.project_name))
```
Comment (jeevanpillay0000) [inline src/app/app.py]: Move into AppGit Run() function

File: src/app/helpers.py [hunk 7]
```diff
@@ -39,3 +42,37 @@ def AppBigQueryStorage():
 
     # return
     return app
+
+
+def CloneRepo():
```
Comment (jeevanpillay0000) [inline src/app/helpers.py]: Rename to RepositoryGit

**Pull request merged**